src/otaclient/create_standby/delta_gen.py

```diff
@@ -0,0 +1,743 @@
+                )
+            )
+            # NOTE: DO NOT CLEANUP the OTA resource folder!
+            if canonical_curdir_path in self.OTA_WORK_PATHS:
+                continue
+
+            if not (
+                canonical_curdir_path == _canonical_root
+                or self._ft_dir_orm.orm_check_entry_exist(
+                    path=str(canonical_curdir_path)
+                )
+            ):
+                dirnames.clear()
+                shutil.rmtree(delta_src_curdir_path, ignore_errors=True)
+                continue
+
+            # NOTE: remove the dir symlinks
+            for _dname in dirnames:
+                _dpath = delta_src_curdir_path / _dname
+                if _dpath.is_symlink():
+                    _dpath.unlink(missing_ok=True)
+
+            for _fname in filenames:
+                _fpath = delta_src_curdir_path / _fname
+                _fpath.unlink(missing_ok=True)
+
+
+class RebuildDeltaWithBaseFileTable(DeltaWithBaseFileTable):
+    """
+    In rebuild mode, base will be the active slot, which we should not modify.
+    """
+
+    def _process_file_thread_worker(self) -> None:
+        """Thread worker to scan files."""
+        worker_helper = ProcessFileHelper(
+            self._que,
+            session_id=self.session_id,
+            report_que=self._status_report_queue,
+        )
+
+        while _input := self._que.get():
+            expected_digest, canonical_fpaths = _input
+            delta_src_fpaths = {
+                Path(
+                    replace_root(
+                        _fpath,
+                        CANONICAL_ROOT,
+                        self._delta_src_mount_point,
+                    )
+                )
+                for _fpath in canonical_fpaths
+            }
+
+            try:
+                for fpath in delta_src_fpaths:
+                    _tmp_fpath = self._copy_dst / _gen_tmp_fname()
+                    try:
+                        calculated_digest, file_size = (
+                            worker_helper.stream_and_verify_file(fpath, _tmp_fpath)
+                        )
+                    except BaseException:
+                        _tmp_fpath.unlink(missing_ok=True)
+                        continue
+
+                    if calculated_digest == expected_digest:
+                        if (
+                            self._rst_orm_pool.orm_delete_entries(
+                                digest=expected_digest
+                            )
+                            == 1
+                        ):
+                            resource_save_dst = self._copy_dst / expected_digest.hex()
+                            os.replace(_tmp_fpath, resource_save_dst)
+                            worker_helper.report_one_file(file_size)
+                        # directly break out once we found a valid resource, no matter
+                        #   we finally need it or not.
+                        break
```

Comment:
When the calculated digest doesn't match the expected digest, nothing will be happened. Is it expected? Or do we need to unlink it?
```
_tmp_fpath.unlink(missing_ok=True)
```

Reply:
Good catch! Let me add an unlink in the else block!